tools/shared.py

```diff
@@ -1743,8 +1743,15 @@ def link(files, target, force_archive_contents=False, temp_files=None, just_calc
     # returns True.
     def consider_object(f, force_add=False):
       new_symbols = Building.llvm_nm(f)
-      if not new_symbols.is_valid():
-        logging.warning('object %s is not valid, cannot link' % (f))
+      # Check if the object was valid according to llvm-nm. It also accepts
+      # native object files.
+      if not new_symbols.is_valid_for_nm():
+        logging.warning('object %s is not valid according to llvm-nm, cannot link' % (f))
+        return False
+      # Check the object is valid for us, and not a native object file.
+      # TODO: for lld, also check if a wasm object file?
```

Comment:
This should be ok with the experimental lld path today right? We're still using bitcode linking pre-llc, so this shouldn't regress anything today. @sbc100 to double-check that.

Reply:
Yes, for now this is fine, once we switch to native object files we will need to modify this, but this is still a good change for now.